lib/bq_stream/oldest_record.rb

```diff
@@ -29,35 +29,43 @@ def self.update_oldest_records_for(table)
         oldest_attr_recs.map(&:bq_earliest_update).reject(&:nil?).uniq.min
       BqStream.log(:info, "#{Time.now}: Table #{table} "\
                    "count #{oldest_attr_recs.count}")
-      next_record = next_record_to_write(table.constantize, earliest_update)
+      next_records = records_to_write(table.constantize, earliest_update)
+      unless next_records.empty?
+        next_records =
+          next_records.where(created_at: next_records.first.created_at)
+      end
```

Comment:
This was pretty confusing to read. I think a cleaner solution would be to move this into the `records_to_write` method, but make the actions more clear.

```ruby
next_records = records_to_write(table.constantize, earliest_update)

def self.records_to_write(table, earliest_update = Time.now)
  # Query all records within our time frame, and grab the first created_at date
  next_created_at = table.where(
    'created_at >= ? AND created_at < ?',
    BqStream.back_date, earliest_update
  ).order('created_at DESC').pluck(:created_at).first

  # Use that to get all records with that date for the current update
  table.where('created_at = ?', next_created_at)
end